controller.go

```diff
@@ -284,6 +319,25 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 
 	network.processOptions(options...)
 
+	// Set proper default address space if not provided
+	if network.ipamConfig.AddressSpace == "" {
+		c.Lock()
+		id, ok := c.ipams[network.ipamType]
+		c.Unlock()
+		if !ok {
+			return nil, types.NotFoundErrorf("could not find ipam driver %s to get default address space", network.ipamType)
+		}
+		global, err := c.isDriverGlobalScoped(network.networkType)
```

Comment:
Here network driver's scope decides the IP address-space to pick (accordingly it's scope). This is correct imo (as usually network-driver's scope will decide addressing scope as well) but its important to point out that it shows how it ties the ipam and networking.

Just stating this out as this seems one example of how addressing relates to networking.